extensions/amp-analytics/0.1/events.js

```diff
@@ -414,6 +421,24 @@ export class VisibilityTracker extends EventTracker {
   }
 
   /**
+   * @return {!Promise}
+   * @visibleForTesting
+   */
+  createReportReadyPromise_() {
+    const viewer = this.root.getViewer();
+
+    if (!viewer.isVisible()) {
+      return Promise.resolve();
+    }
+
+    return new Promise(resolve => {
+      viewer.onVisibilityChanged(() => {
+        resolve();
```

Comment:
Let's keep the check of `!viewer.isVisible()` before calling resolve. It's possible that visibility changed is called many times w/o actually changing `isVisible` boolean.